hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/request/EdgeRequestBuilder.java

```diff
@@ -49,40 +54,44 @@ class EdgeRequestBuilder {
       ArgumentDeserializer argumentDeserializer,
       GraphQlSelectionFinder selectionFinder,
       MetricAggregationRequestBuilder metricAggregationRequestBuilder,
+      FilterRequestBuilder filterRequestBuilder,
       AttributeRequestBuilder attributeRequestBuilder,
       Provider<NeighborEntitiesRequestBuilder> neighborEntitiesRequestBuilderProvider) {
     this.argumentDeserializer = argumentDeserializer;
     this.selectionFinder = selectionFinder;
     this.metricAggregationRequestBuilder = metricAggregationRequestBuilder;
     this.attributeRequestBuilder = attributeRequestBuilder;
     this.neighborEntitiesRequestBuilderProvider = neighborEntitiesRequestBuilderProvider;
+    this.filterRequestBuilder = filterRequestBuilder;
   }
 
   Single<EdgeSetGroupRequest> buildIncomingEdgeRequest(
       GraphQlRequestContext context,
       TimeRangeArgument timeRange,
       Optional<String> space,
       Stream<SelectedField> edgeSetFields) {
-    return this.buildEdgeRequest(
-        context, timeRange, space, this.getEdgesByType(edgeSetFields), EdgeType.INCOMING);
+    return this.buildEdgeRequest(context, timeRange, space, edgeSetFields, EdgeType.INCOMING);
   }
 
   Single<EdgeSetGroupRequest> buildOutgoingEdgeRequest(
       GraphQlRequestContext context,
       TimeRangeArgument timeRange,
       Optional<String> space,
       Stream<SelectedField> edgeSetFields) {
-    return this.buildEdgeRequest(
-        context, timeRange, space, this.getEdgesByType(edgeSetFields), EdgeType.OUTGOING);
+    return this.buildEdgeRequest(context, timeRange, space, edgeSetFields, EdgeType.OUTGOING);
   }
 
   private Single<EdgeSetGroupRequest> buildEdgeRequest(
       GraphQlRequestContext context,
       TimeRangeArgument timeRange,
       Optional<String> space,
-      Map<String, Set<SelectedField>> edgesByType,
+      Stream<SelectedField> edgeSetFields,
```

Comment:
This looks to be off. There are potentially multiple edge requests that each can have different fields and filters - hence the previous map of type to fields. We're splitting them up later here, but we're not doing the same thing for filters.